File: ScaleformUI_V8/src/menus/UIMenu/sidepanels/ColorPicker/uivehiclecolourpickerpanel.ts
import { ScaleformUI } from "scaleforms/scaleformui/main";
import { PanelSide, SidePanelsTitleType, UIMenuSidePanel } from "../uimenusidepanel";
import { VehicleColorPickerSelectEvent, VehicleColorPickerSelectEventBuilder } from '../../emitters/emitters';
import { SColor } from "elements/scolor";

export class UIVehicleColourPickerPanel extends UIMenuSidePanel {
    private title: string;
    public get Title(): string {
        return this.title;
    }
    public set Title(value: string) {
        this.title = value;
        if (this.ParentItem != null && this.ParentItem.Parent != null && this.ParentItem.Parent.Visible) {
            ScaleformUI.Scaleforms._ui?.callFunction("UPDATE_SIDE_PANEL_TITLE", this.ParentItem.Parent.Pagination.GetScaleformIndex(this.ParentItem.Parent.Items.indexOf(this.ParentItem)), this.title);
        }
    }

    public TitleColor: SColor;
    _titleType: SidePanelsTitleType;
    _value: number;

    public get Value():number { return this._value; }
    private _VehicleColorPickerSelect = new VehicleColorPickerSelectEventBuilder();
    public OnVehicleColorPickerSelect(delegate: VehicleColorPickerSelectEvent) {
        this._VehicleColorPickerSelect.add(delegate);
    }
    PickerSelect() {
        this._VehicleColorPickerSelect.toDelegate()(this.ParentItem, this, this._value)
    }

    constructor(side: PanelSide, title: string, titleColor: SColor) {
        super();
        this.PanelSide = side;
        this.title = title;
        this.TitleColor = titleColor;
        this._titleType = titleColor == SColor.HUD_None ? SidePanelsTitleType.Big : SidePanelsTitleType.Small;
        this._value = 0;
    }
}

File: ScaleformUI_Csharp/Menus/UIMenu/SidePanels/ColorPicker/UIVehicleColourPickerPanel.cs
using ScaleformUI.Elements;
using System.Drawing;

namespace ScaleformUI.Menu
{

    public class UIVehicleColourPickerPanel : UIMenuSidePanel
    {
        private string title;
        public string Title
        {
            get => title;
            set
            {
                title = value;
                if (ParentItem is not null && ParentItem.Parent != null && ParentItem.Parent.Visible)
                {
                    int it = ParentItem.Parent.MenuItems.IndexOf(ParentItem);
                    ParentItem.Parent.SendSidePanelToScaleform(it, true);
                }
            }
        }
        public SColor TitleColor;
        internal SColor color;
        internal SidePanelsTitleType _titleType;
        internal int _value;

        public int Value => _value;
        public SColor Color => color;
        public event VehicleColorPickerSelectEvent OnVehicleColorPickerSelect;
        public event VehicleColorPickerHoverEvent OnColorHover;

        /// <summary>
        /// Adds a Mission Details panel as side menu panel
        /// </summary>
        /// <param name="side">Left or Right?</param>
        /// <param name="title">Panel's title</param>
        /// <param name="inside">if true the title will be within the panel itself else will be the same as the menu.</param>
        /// <param name="txd">Texture dictionary for the picture</param>
        /// <param name="txn">Texture name for the picture</param>
        public UIVehicleColourPickerPanel(PanelSide side, string title)
        {
            PanelSide = side;
            _titleType = SidePanelsTitleType.Big;
            Title = title;
            TitleColor = SColor.HUD_None;
        }

        /// <summary>
        /// Adds a Mission Details panel as side menu panel
        /// </summary>
        /// <param name="side">Left or Right?</param>
        /// <param name="title">Panel's title</param>
        /// <param name="color">Background color for the panel title</param>
        public UIVehicleColourPickerPanel(PanelSide side, string title, SColor titleColor)
        {
            PanelSide = side;
            _titleType = SidePanelsTitleType.Small;
            Title = title;
            TitleColor = titleColor;
        }

        internal void PickerSelect(SColor color)
        {
            this.color = color;
            OnVehicleColorPickerSelect?.Invoke(ParentItem, this, _value, color);
        }
        internal void PickerHovered(int colorId, SColor color)
        {
            OnColorHover?.Invoke(colorId, color);
        }
        internal void PickerRollout()
        {
            OnColorHover?.Invoke(_value, color);
        }
    }
}
